File: services/orchestrator/src/util/redact.ts
// TS mirror of the Go sandbox redactCreds (defense-in-depth on the consumer
// side): strips URL userinfo and bare credentials so secrets never appear in
// orchestrator-thrown error messages, logs, or responses.
const URL_CREDS = /([a-zA-Z][a-zA-Z0-9+.\-]*:\/\/)[^/@\s]+@/g;
const BARE = [
  /x-access-token:[^@\s/]+/gi,
  /gh[pousr]_[0-9A-Za-z]+/g,
  /github_pat_[0-9A-Za-z_]+/g,
  /(?:Bearer|token)\s+[^\s,;]+/gi,
  /AKIA[0-9A-Z]{16}/g,
];

export function redactCreds(s: string): string {
  let out = s.replace(URL_CREDS, "$1[redacted]@");
  for (const re of BARE) out = out.replace(re, "[redacted]");
  return out;
}

File: services/sandbox-runner/internal/sandbox/redact.go
package sandbox

import "regexp"

// matches scheme://userinfo@  (userinfo = anything up to the @, no slash/space)
var urlCredsRe = regexp.MustCompile(`([a-zA-Z][a-zA-Z0-9+.\-]*://)[^/@\s]+@`)

// redactCreds removes userinfo (user:pass@ or token@) from any URLs in s,
// so credentials never appear in errors, logs, or HTTP responses.
func redactCreds(s string) string {
	return urlCredsRe.ReplaceAllString(s, "${1}[redacted]@")
}
